src/modules/launcher/Wox.Test/FuzzyMatcherTest.cs

```diff
@@ -122,21 +122,37 @@ public void WhenGivenStringsAndAppliedPrecisionFilteringThenShouldReturnGreaterT
             }
         }
 
-        [TestCase(Chrome, Chrome, 137)]
-        [TestCase(Chrome, LastIsChrome, 83)]
-        [TestCase(Chrome, HelpCureHopeRaiseOnMindEntityChrome, 21)]
-        [TestCase(Chrome, UninstallOrChangeProgramsOnYourComputer, 15)]
-        [TestCase(Chrome, CandyCrushSagaFromKing, 0)]
-        [TestCase("sql", MicrosoftSqlServerManagementStudio, 56)]
-        [TestCase("sql  manag", MicrosoftSqlServerManagementStudio, 79)]//double spacing intended
-        public void WhenGivenQueryStringThenShouldReturnCurrentScoring(string queryString, string compareString, int expectedScore)
+        [Test]
+        public void WhenMultipleResults_ExactMatchingResult_ShouldHaveGreatestScore()
         {
-            // When, Given
+            // Arrange
+            string queryString = "vim";
+
+            // Vim
+            string firstName = "Vim";
+            string firstDescription = "Vi Improved - A Text Editor";
+            string firstExecutableName = "vim.exe";
+
+            // Vim Diff
+            string secondName = "Vim Diff";
+            string secondDescription = "Vi Improved - A Text Editor";
+            string secondExecutableName = "vim.exe";
+
+            // Act
             var matcher = new StringMatcher();
-            var rawScore = matcher.FuzzyMatch(queryString, compareString).RawScore;
+            var firstNameMatch = matcher.FuzzyMatch(queryString, firstName).RawScore;
+            var firstDescriptionMatch = matcher.FuzzyMatch(queryString, firstDescription).RawScore;
+            var firstExecutableNameMatch = matcher.FuzzyMatch(queryString, firstExecutableName).RawScore;
 
-            // Should
-            Assert.AreEqual(expectedScore, rawScore, $"Expected score for compare string '{compareString}': {expectedScore}, Actual: {rawScore}");
+            var secondNameMatch = matcher.FuzzyMatch(queryString, secondName).RawScore;
+            var secondDescriptionMatch = matcher.FuzzyMatch(queryString, secondDescription).RawScore;
+            var secondExecutableNameMatch = matcher.FuzzyMatch(queryString, secondExecutableName).RawScore;
+
+            var firstScore = new[] { firstNameMatch, firstDescriptionMatch, firstExecutableNameMatch }.Max();
+            var secondScore = new[] { secondNameMatch, secondDescriptionMatch, secondExecutableNameMatch }.Max();
```

Comment:
[clarification] `DescriptionMatch`, `ExecutableNameMatch` will have the same score in both first and second cases and would be ignored by the max operation. Is it required?

Reply:
Yes, I agree. The reason I did it is to emulate the way scoring is done in the code for Win32 Applications. Rather than just comparing the names, which have the max value, I also wanted to show that even though the exe has the same name as the query string, it would still order the results correctly and not affect the scoring, ie. the one with the exact match would have a higher score. However, for clarity, I shall rename it to ignore.exe as suggested by Ryan.

Reply:
Sounds good!! Thanks!